purchase_propagate_qty/tests/test_purchase_delivery.py

```diff
@@ -94,3 +94,57 @@ def test_purchase_line_qty_decrease_to_zero(self):
         line1.write({"product_qty": 0})
         self.assertEqual(move1.product_uom_qty, 0)
         self.assertEqual(move1.state, "cancel")
+
+    def test_purchase_line_splitted_move(self):
+        # Add qty, create a new move with the diff
+        line1 = self.po.order_line[0]
+        move1 = self.env["stock.move"].search([("purchase_line_id", "=", line1.id)])
+        self.assertEqual(len(move1), 1)
+        line1.write({"product_qty": 64})
+        self.assertEqual(move1.product_uom_qty, 64)
+        # split the move
+        move1._split(22)
```

Comment:
Why not split the move the way Odoo does it, by changing the qty + the price unit of the PO line? To stick to the standard workflow.